plugins/woocommerce/src/Internal/FraudProtection/CartEventTracker.php

```diff
@@ -0,0 +1,299 @@
+	 *
+	 * @internal
+	 *
+	 * @param SessionDataCollector      $data_collector              The session data collector instance.
+	 * @param SessionClearanceManager   $session_clearance_manager   The session clearance manager instance.
+	 * @param FraudProtectionController $fraud_protection_controller The fraud protection controller instance.
+	 */
+	final public function init(
+		SessionDataCollector $data_collector,
+		SessionClearanceManager $session_clearance_manager,
+		FraudProtectionController $fraud_protection_controller
+	): void {
+		$this->data_collector              = $data_collector;
+		$this->session_clearance_manager   = $session_clearance_manager;
+		$this->fraud_protection_controller = $fraud_protection_controller;
+	}
+
+	/**
+	 * Register cart event hooks.
+	 *
+	 * Hooks into WooCommerce cart actions to track fraud protection events.
+	 * Only registers hooks if the fraud protection feature is enabled.
+	 *
+	 * @return void
+	 */
+	public function register(): void {
+		// Only register hooks if fraud protection is enabled.
+		if ( ! $this->fraud_protection_controller->feature_is_enabled() ) {
+			return;
+		}
+
+		add_action( 'woocommerce_add_to_cart', array( $this, 'handle_track_cart_item_added' ), 10, 6 );
+		add_action( 'woocommerce_after_cart_item_quantity_update', array( $this, 'handle_track_cart_item_updated' ), 10, 4 );
+		add_action( 'woocommerce_remove_cart_item', array( $this, 'handle_track_cart_item_removed' ), 10, 2 );
+		add_action( 'woocommerce_restore_cart_item', array( $this, 'handle_track_cart_item_restored' ), 10, 2 );
+	}
+
+	/**
+	 * Handle cart item added event.
+	 *
+	 * Triggers fraud protection event tracking when an item is added to the cart.
+	 *
+	 * @internal
+	 *
+	 * @param string $cart_item_key  Cart item key.
+	 * @param int    $product_id     Product ID.
+	 * @param int    $quantity       Quantity added.
+	 * @param int    $variation_id   Variation ID.
+	 * @param array  $variation      Variation data.
+	 * @param array  $cart_item_data Cart item data.
+	 * @return void
+	 */
+	public function handle_track_cart_item_added( $cart_item_key, $product_id, $quantity, $variation_id, $variation, $cart_item_data ): void {
+		$event_data = $this->build_cart_event_data(
+			'item_added',
+			$product_id,
+			$quantity,
+			$variation_id
+		);
+
+		$this->track_event( 'cart_item_added', $event_data );
+	}
+
+	/**
+	 * Handle cart item quantity updated event.
+	 *
+	 * Triggers fraud protection event tracking when cart item quantity is updated.
+	 *
+	 * @internal
+	 *
+	 * @param string $cart_item_key Cart item key.
+	 * @param int    $quantity      New quantity.
+	 * @param int    $old_quantity  Old quantity.
+	 * @param object $cart          Cart object.
+	 * @return void
+	 */
+	public function handle_track_cart_item_updated( $cart_item_key, $quantity, $old_quantity, $cart ): void {
+		$cart_item = isset( $cart->cart_contents[ $cart_item_key ] ) ? $cart->cart_contents[ $cart_item_key ] : null;
```

Comment:
What do you think of using the null coalescing operator here (and elsewhere) instead of the issets, like:

```suggestion
		$cart_item = $cart->cart_contents[ $cart_item_key ] ?? null;
```

Reply:
Fixed with 60e694299bc02491513758282d1c17dd134a8971